clinica/pipelines/t1_freesurfer_longitudinal/t1_freesurfer_longitudinal_correction_utils.py

```diff
@@ -0,0 +1,856 @@
+        out_subject_list,
+        out_session_list,
+        out_capstarget_list,
+        out_overwrite_warning] = utils.to_process(
+            in_pcssd_capstargetlist,
+            unpcssd_sublist,
+            unpcssd_seslist,
+            unpcssd_capstargetlist,
+            all_sublist,
+            all_seslist,
+            all_capstargetlist,
+            in_caps_dir,
+            in_overwrite_caps,
+            in_working_directory,
+            in_n_procs,
+            in_overwrite_tsv)
+
+    # copy in_caps_dir, in_overwrite_caps
+    out_caps_dir = in_caps_dir
+    out_overwrite_caps = in_overwrite_caps
+
+
+    return [
+        out_subject_list,
+        out_session_list,
+        out_capstarget_list,
+        out_caps_dir,
+        out_overwrite_warning,
+        out_overwrite_caps]
+
+
+
+def create_symlinks(
+        in_caps_dir,
+        in_subject_list,
+        in_session_list,
+        in_longsubdirnames_dict):
+    """Create symlinks for recon-all execution
+
+    t1-freesurfer-cross-sectional has created the following folders:
+    [caps_dir]/subjects/sub-[sub_i]/ses-[ses_ij]/t1/...
+    ...freesurfer_cross_sectional/sub-[sub_i]_ses-[ses_ij]
+    and t1-freesurfer-template has created:
+    [caps_dir]/subjects/sub-[sub_i]/long-[long_i]/...
+    ...freesurfer_unbiased_template/sub-[sub_i]
+    where i in range [1, subject_number], [long_i] corresponds to all
+    seessions associated with [sub_i] and ij represents any of the
+    potential sessions for subject i
+    This function creates for all above folders the following symbolic
+    link ./subjects/sub-[sub_i]_ses-[ses_ij]
+    and
+    ./subjects/sub-[sub_i] (template) in the working directory.
+    All the symlinks are located in the same folder, disrespective of
+    the subject (i.e. use a Node and not a MapNode). The path of that
+    folder is returned so recon-all -base can later be run on all the
+    sessions corresponding to subject i.
+
+    Args:
+        in_caps_dir (string): CAPS directory to contain the output
+        in_subject_list (list of string): list of participant_id
+        in_session_list (list of string): list of corresponding
+            session_id
+        in_longsubdirnames_dict (dictionary of strings): for each
+            unique subject, name of longitudinal subofolder associated
+            to the current set of visits
+
+    Returns:
+        out_subject_symlink_path (string): path to the folder
+            containing all session symlinks and all template symlinks
+            for all subject_id
+    """
+
+    import os
+
+    # change the relative path to be absolute path
+    caps_path = os.path.expanduser(in_caps_dir)
+    caps_dir = os.path.join(caps_path, 'subjects')
+
+    # create subject-specific path in clinica's working directory
+    # (later used as $SUBJECT_DIR)
+    out_subject_symlink_path = os.path.abspath('./subjects')
+    try:
+        os.mkdir(out_subject_symlink_path)
+    except OSError:
+        pass
```

Comment:
exception levée  = dossier existe déjà ?

Reply:
I suspect I got this from stackoverflow (can't find the original page though).
The idea would be that if we run the pipeline once and that it crashes at node 3 ('3_symlink', corresponding utils: create_symlinks()) after the folder has been created (but before the function completes), we can re-run the pipeline and the code won't complain about the folder having already been created.

I reckon we could also go for something simpler like
```
if not os.path.exists(out_subject_symlink_path):
    os.mkdir(out_subject_symlink_path)
```

Reply:
Changed as per the top answer of https://stackoverflow.com/questions/273192/how-can-i-safely-create-a-nested-directory